### PR TITLE
Stop using BytesSize when merging the config

### PR DESCRIPTION
**Pull request by bitoku**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When you provide `--runtime` option (or `CONTAINER_RUNTIMES` variable) with no `container_min_memory`, it doesn't respect the default container_min_memory (12MiB) and it will be 0B. 

```
sudo crio --log-level=debug --runtimes=runc:/usr/local/sbin/runc:

...
INFO[2024-04-22 12:17:08.039694857Z] Using runtime handler runc version 1.1.12, commit: v1.1.12-0-g51d5e946, spec: 1.0.2-dev, go: go1.20.13, libseccomp: 2.5.4  file="config/config.go:1291"
DEBU[2024-04-22 12:17:08.039769758Z] Runtime handler "runc" container minimum memory set to 0 bytes  file="config/config.go:1300"
...
```

It should be 12 MiB.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by saschagrunert — COMMENTED** [review]
@roman-kiselenko PTAL

I don't think we need a release note for this @bitoku because we have not released the feature yet. Thank you for the PR!

File: test/image.bats
Comment (roman-kiselenko) [inline test/image.bats:528]: Can we check if the log file contains the line about the `container_min_memory` default value?
Something like:
```sh
wait_for_log "Runtime handler \"runc\" container minimum memory set to 12582912 bytes"
```

This can supplement the test with the necessary verification. 🤔

**Review by roman-kiselenko — APPROVED** [review]
Nothing is blocking it, just small test improvements.

File: test/image.bats
Comment (haircommander) [inline test/image.bats:546]: `run` here won't report whether the command failed or not, it'll save it in variables. I recommend dropping `run` if we expect this command to fail

Comment (bitoku) [reply test/image.bats:546]: according to the doc https://bats-core.readthedocs.io/en/stable/writing-tests.html#when-not-to-use-run,

> Note: In contrast to the above, testing that a command failed is best done via
`run ! command args ...`

However I think just checking the log line is enough as @/roman-kiselenko said.

I'm going to remove this test and add log check to the existing test.

File: test/image.bats
Comment (roman-kiselenko) [inline test/image.bats:460]: I think there is a typo or something; this test is supposed to run [a container and doesn't fail.](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/cri-o_cri-o/8057/pull-ci-cri-o-cri-o-main-ci-fedora-kata/1782433161666367488/artifacts/fedora-kata/cri-o-gather/artifacts/testout.txt#:~:text=alpine%40sha256%3A65b3a80ebe7471beecbc090c5b2cdd0aafeaefa0715f8f12e40dc918a3a70e32-,%23%20f0b894c92ead47bdeffc3e64066d8631b22bfa7edc679156b8843283acf82d6f,-%23%20%23%20%2D%2D%2D%20crio.log%20%3A%3A%20%2D%2D%2D%0A%23%20time) 

`container_min_memory` is `"17.5MiB"` and it is enough to launch a container.

Comment (bitoku) [reply test/image.bats:460]: container_min_memory is a minimum memory so the container memory should be above the minimum, right?
https://github.com/cri-o/cri-o/blob/9bac613f5fd24c3d690dfb2233f600d157e94baa/internal/config/cgmgr/cgmgr_linux.go#L149-L155

It sets 12MiB as a container memory so it should be failed, shouldn't it?

https://github.com/cri-o/cri-o/blob/c65ecf08ccd32b90e4fcf97b8f5c46418904c506/test/image.bats#L456

Comment (roman-kiselenko) [reply test/image.bats:460]: This function does the check here:
https://github.com/cri-o/cri-o/blob/d90763d245d0756977cf4db0f9b8c0bbf2796f8f/server/container_create_linux.go#L434-L436 

- `memoryLimit` - `resources.memory_limit_in_bytes` memory requested for container
- `minMemory`   - `container_min_memory` minimum memory needs to run a container

```go
func VerifyMemoryIsEnough(memoryLimit /* 12MiB */, minMemory /* 17MiB */) error { 
  if 12MiB < 17MiB {
    return error
  }
  // ...
}
```

Hence, if `memoryLimit` is above `minMemory`, the container can't start.

Comment (roman-kiselenko) [reply test/image.bats:460]: @cri-o/cri-o-maintainers ^^^ WDYT?

Comment (bitoku) [reply test/image.bats:460]: sorry I can't still get it.
12 MiB < 17 MiB is true, so `VerifyMemoryIsEnough` should return an error and the container can't start...?

the same as https://github.com/cri-o/cri-o/pull/7832/files#diff-6391e62911e38278fada73fcfa52f4480e77cb2aea66483934da7e66871dddb5

**Comment by bitoku** [comment]
It seems crio doesn't respect the default runtime `mem` set in conf because of this.
The test fails because container_min_memory in `containerd-shim-kata-v2` is default (12 MiB).

https://github.com/cri-o/cri-o/blob/08c7f32cdc0fa121738154fd8bf05195dd97e1a0/contrib/test/ci/vars.yml#L81

**Comment by bitoku** [comment]
Can we skip the tests for kata in the same way as the ones written in the vars.yml?

File: test/image.bats
Comment (roman-kiselenko) [reply test/image.bats:460]: Oh, I see, you're right `run ! ...` is the right assertion here.

**Review by roman-kiselenko — APPROVED** [review]
/approve

**Comment by haircommander** [comment]
hm
```
# (from function `crictl' in file ./helpers.bash, line 87,
#  in test file ./image.bats, line 503)
#   `crictl pull ${IMAGE_LIST_DIGEST_FOR_TAG_AMD64}' failed
# time="2024-04-22 16:42:21.070657374Z" level=info msg="Starting CRI-O, version: 1.30.0, git: 837db7e253aac215e7f3d6b49665567dbe80d71d(clean)"
# level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL"
# time="2024-04-22 16:42:21.120060555Z" level=info msg="Starting CRI-O, version: 1.30.0, git: 837db7e253aac215e7f3d6b49665567dbe80d71d(clean)"
# level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL"
# time="2024-04-22T16:42:21Z" level=fatal msg="validate service connection: validate CRI v1 runtime API for endpoint \"unix:///tmp/tmp.zsmL7eIvKa/crio.sock\": rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial unix /tmp/tmp.zsmL7eIvKa/crio.sock: connect: no such file or directory\""
# E0422 16:43:22.568155   67612 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = can't talk to a V1 container registry" image="quay.io/crio/alpine@sha256:65b3a80ebe7471beecbc090c5b2cdd0aafeaefa0715f8f12e40dc918a3a70e32"
# time="2024-04-22T16:43:22Z" level=fatal msg="pulling image: can't talk to a V1 container registry"
```
/hold

**Comment by kwilczynski** [comment]
Some of the integration tests fail on Arm64 with:

```console
/home/runner/actions-runner/_work/cri-o/cri-o/test/./image.bats: line 501: go: command not found
```

Go not installed? Not in `PATH`?

**Comment by bitoku** [comment]
There hasn't been Go cli in the arm64 runner all along, but because crictl was wrapped with `run` which ignored the failure, the tests didn't fail.
I changed `go env GOARCH` to use `ARCH`.

File: contrib/test/ci/vars.yml
Comment (roman-kiselenko) [inline contrib/test/ci/vars.yml:155]: Can we skip only the failing test? WDYT?

Comment (kwilczynski) [reply contrib/test/ci/vars.yml:155]: Which one is it? Why only a single one would fail, but not all of them for Kata? :thinking:

Comment (roman-kiselenko) [reply contrib/test/ci/vars.yml:155]: AFAIK only this test fails `run container with container_min_memory 17.5MiB"`

Comment (bitoku) [reply contrib/test/ci/vars.yml:155]: For kata, the runtime settings written in the test cases are not used, but the kata runtime is used instead. (the settings are specified in vars.yml)

In the settings for kata, it doesn't set `container_min_memory`  explicitly so it will be 12MiB (default).

In the 17.5 MiB test, normally it is expected to be failed because the default runtime   `container_min_memory` is supposed to have 17.5 MiB but the container's memory limit is 12MiB (12MiB < 17.5 MiB).
However for kata, kata's `container_min_memory` is 12MiB and it succeeds in creating the container, which caused the test failure.

In my opinion, the other tests just happen to have the same result in both non-kata and kata, so it should be removed from the kata tests.
Also it's confusing because we might think the runtime set in the test cases are used although they are not used.

**Pull request merged**
